ci/travis/ci.sh

```diff
@@ -127,24 +313,90 @@ build() {
+        local MOUNT_BAZEL_CACHE=(-v "${HOME}/ray-bazel-cache":/root/ray-bazel-cache -e TRAVIS=true -e TRAVIS_PULL_REQUEST="${TRAVIS_PULL_REQUEST:-false}" -e encrypted_1c30b31fe1ee_key="${encrypted_1c30b31fe1ee_key-}" -e encrypted_1c30b31fe1ee_iv="${encrypted_1c30b31fe1ee_iv-}")
+
+        # This command should be kept in sync with ray/python/README-building-wheels.md,
+        # except the "${MOUNT_BAZEL_CACHE[@]}" part.
+        suppress_output docker run --rm -w /ray -v "${PWD}":/ray "${MOUNT_BAZEL_CACHE[@]}" -e TRAVIS_COMMIT="${TRAVIS_COMMIT}" rayproject/arrow_linux_x86_64_base:python-3.8.0 /ray/python/build-wheel-manylinux1.sh
+      fi
+      ;;
+    darwin*)      
+      if [ "${MAC_WHEELS-}" = 1 ]; then
+        # This command should be kept in sync with ray/python/README-building-wheels.md.
+        suppress_output "${WORKSPACE_DIR}"/python/build-wheel-macos.sh
+      fi
+      ;;
+    msys*)
+      if [ "${WINDOWS_WHEELS-}" = 1 ]; then
+        (
+          cd "${WORKSPACE_DIR}"/python
+          python setup.py --quiet bdist_wheel
+        )
+      fi
+      ;;
+  esac
+}
+
+run() {
+  local result=0 flush_logs=0
+  ##### BEGIN TASKS #####
+
+  if [ "${LINT-}" = 1 ]; then
+    lint
+  fi
+
+  if [ "${RAY_DEFAULT_BUILD-}" = 1 ]; then
+    test_python
+  fi
+
+  if [ "${RAY_INSTALL_JAVA-}" = 1 ]; then
+    "${WORKSPACE_DIR}"/java/test.sh
+  fi
 
-    # This command should be kept in sync with ray/python/README-building-wheels.md,
-    # except the "${MOUNT_BAZEL_CACHE[@]}" part.
-    suppress_output docker run --rm -w /ray -v "${PWD}":/ray "${MOUNT_BAZEL_CACHE[@]}" -e TRAVIS_COMMIT="${TRAVIS_COMMIT}" -ti rayproject/arrow_linux_x86_64_base:python-3.8.0 /ray/python/build-wheel-manylinux1.sh
+  if [ "${TESTSUITE-}" = cpp_worker ]; then
+    bazel test --config=ci //cpp:all --build_tests_only --test_output=streamed
   fi
 
-  if [ "${MAC_WHEELS-}" = 1 ]; then
-    # This command should be kept in sync with ray/python/README-building-wheels.md.
-    suppress_output "${WORKSPACE_DIR}"/python/build-wheel-macos.sh
+  local should_test_wheels=""
+  case "${OSTYPE}" in
+    linux*) should_test_wheels="${LINUX_WHEELS-}";;
+    darwin*) should_test_wheels="${MAC_WHEELS-}";;
+  esac
+  if [ "${should_test_wheels}" = 1 ]; then
+    "${WORKSPACE_DIR}"/ci/travis/test-wheels.sh || { result=$? && flush_logs=1; }
+  fi
+
+  ##### END TASKS #####
+  if [ 0 -ne "${flush_logs}" ]; then
+    local f
+    for f in /tmp/ray/session_latest/logs/*; do
+      if [ -f "$f" ]; then  # make sure the wildcard actually expanded to something valid
+        cat -- "$f"
+      fi
+    done
+    sleep 60  # Explicitly sleep 60 seconds for logs to go through
+  fi
+  return "${result}"
+}
+
+_main() {
+  if [ -n "${GITHUB_WORKFLOW-}" ]; then
+    # Necessary for GitHub Actions (which uses separate shells for different commands)
+    # Unnecessary for Travis (which uses one shell for different commands)
+    reload_env
+  fi
+  if [ 0 -lt "$#" ]; then
```

Comment:
what does this do?

Reply:
It's supposed to check if any arguments were provided before attempting to execute the provided command, but it might be necessary now, I can probably remove it.